msu/hooks/states/world_state.nut

```diff
@@ -200,40 +200,93 @@
 	{
 		onBeforeSerialize(_out);
 		local meta = _out.getMetaData();
+		local modIDsString = ""
 		foreach (mod in ::MSU.System.Serialization.Mods)
 		{
 			meta.setString(mod.getID() + "Version", mod.getVersionString());
 			::MSU.Mod.Debug.printLog(format("MSU Serialization: Saving %s (%s), Version: %s", mod.getName(), mod.getID(), mod.getVersionString()));
 		}
+		foreach (mod in ::mods_getRegisteredMods()) modIDsString += mod.Name + ",";
+		meta.setString("MSU.SavedModIDs", modIDsString.slice(0, -1))
 	}
 
 	local onBeforeDeserialize = o.onBeforeDeserialize;
 	o.onBeforeDeserialize = function( _in )
 	{
 		onBeforeDeserialize(_in);
-		foreach (mod in ::MSU.System.Serialization.Mods)
+
+		if (::MSU.Mod.Serialization.isSavedVersionAtLeast("1.1.0", _in.getMetaData()))
 		{
-			local oldVersion = _in.getMetaData().getString(mod.getID() + "Version");
-			if (oldVersion == "")
+			local modIDs = split(_in.getMetaData().getString("MSU.SavedModIDs"), ",");
+			local hooksMods = ::mods_getRegisteredMods();
+			foreach (mod in hooksMods)
 			{
-				::logInfo(format("MSU Serialization: First time loading this save with %s (%s)", mod.getName(), mod.getID()));
+				local IDIdx = modIDs.find(mod.Name);
+				if (IDIdx != null)
+				{
+					modIDs.remove(IDIdx);
+					if (::MSU.System.Registry.hasMod(mod.Name))
+					{
+						local oldVersion = _in.getMetaData().getString(mod.Name + "Version");
+						if (oldVersion == "")
+						{
+							::logInfo(format("MSU Serialization: First time time this save has been loaded with an MSU version of %s (%s)", mod.FriendlyName, mod.Name));
```

Comment:
Typo: `time time`. Also write `of mod %s` instead of `of %s`.

Reply:
Nope, I've specifically changed away from that in these because it can look weird if eg it's a DLC or if the mod already had mod in the name (Legends)